packages/widget/src/pages/ErrorPage/ErrorPageBadPriceWarning.tsx

```diff
@@ -32,11 +32,22 @@ export const ErrorPageBadPriceWarning = ({
     destAssetChainID,
   } = route;
 
-  const swapDifferencePercentage = `${calculatePercentageChange(
-    usdAmountIn ?? 0,
-    usdAmountOut ?? 0,
-    true,
-  )}%`;
+  const hasUsdValues =
+    usdAmountIn &&
+    usdAmountOut &&
+    parseFloat(usdAmountIn) > 0 &&
+    parseFloat(usdAmountOut) > 0;
+
+  const swapDifferencePercentage = hasUsdValues
+    ? `${calculatePercentageChange(usdAmountIn, usdAmountOut, true)}%`
+    : null;
+
+
+  const priceImpactPercentage = (() => {
+    const impactString = route.swapPriceImpactPercent;
+    if (!impactString) return null;
+    return `${parseFloat(impactString).toFixed(2)}%`;
+  })();
```

Comment:
i'd avoid using these immediately-invoked functions, and use `useMemo` instead, something like this

```suggestion
  const priceImpactPercentage = useMemo(() => {
    const impactString = route.swapPriceImpactPercent;
    if (!impactString) return null;
    return `${parseFloat(impactString).toFixed(2)}%`;
  }), [route.swapPriceImpactPercent]);
```